src/H5VLint.c

```diff
@@ -113,26 +115,29 @@ static herr_t         H5VL__free_vol_wrapper(H5VL_wrap_ctx_t *vol_wrap_ctx);
 
 /* VOL ID class */
 static const H5I_class_t H5I_VOL_CLS[1] = {{
-    H5I_VOL,                   /* ID class value */
-    0,                         /* Class flags */
-    0,                         /* # of reserved IDs for class */
-    (H5I_free_t)H5VL__free_cls /* Callback routine for closing objects of this class */
+    H5I_VOL,                       /* ID class value */
+    0,                             /* Class flags */
+    0,                             /* # of reserved IDs for class */
+    (H5I_free_t)H5VL__conn_free_id /* Callback routine for closing objects of this class */
 }};
 
 /* Declare a free list to manage the H5VL_class_t struct */
 H5FL_DEFINE_STATIC(H5VL_class_t);
 
-/* Declare a free list to manage the H5VL_t struct */
-H5FL_DEFINE(H5VL_t);
+/* Declare a free list to manage the H5VL_connector_t struct */
+H5FL_DEFINE_STATIC(H5VL_connector_t);
 
 /* Declare a free list to manage the H5VL_object_t struct */
 H5FL_DEFINE(H5VL_object_t);
 
 /* Declare a free list to manage the H5VL_wrap_ctx_t struct */
 H5FL_DEFINE_STATIC(H5VL_wrap_ctx_t);
 
+/* List of currently active VOL connectors */
+static H5VL_connector_t *H5VL_conn_list_head_g = NULL;
```

Comment:
Keeps a searchable list of the connectors that have been registered.  Should be very short, so not worried about linear search time for the couple of H5VL API routines that search the list.